repoclosure/repoclosure.sh

```diff
@@ -13,13 +13,15 @@ shift; shift
 
 TEMPDIR=$(mktemp -d)
 trap "rm -rf $TEMPDIR" EXIT
+MODULESDIR=$TEMPDIR/modules.d
 
 # repo names must be unique, or yum will get confused between different OSes and URLs
 reponame=undertest-$(basename $yumorig .conf)-$(echo $url | cksum | sed 's/ /-/g')
 
 yumconf=$TEMPDIR/yum.conf
 cat $yumorig > $yumconf
 cat >> $yumconf << EOF
+modulesdir=$MODULESDIR
```

Comment:
I saw this in https://github.com/rpm-software-management/libdnf/blob/eff7e688e2f1ad5d10a5ed05948c08141083f316/data/tests/modules/etc/dnf/dnf.conf.in#L6 but looking at https://github.com/rpm-software-management/libdnf/blob/dnf-4-master/libdnf/module/ModulePackageContainer.cpp it is hardcoded to `/etc/dnf/modules.d`. I think we have to switch to containers for proper modularity repoclosure.

Reply:
Looking at https://github.com/rpm-software-management/libdnf/blob/eff7e688e2f1ad5d10a5ed05948c08141083f316/libdnf/module/ModulePackageContainer.cpp#L297-L298, you should be able to pass `--installroot` to the DNF call and point it at a different "root" that has an `/etc/dnf/modules.d`

Reply:
Yep, that works. In that case you must also pass `--releasever`. However, that info is also quite useful to have anyway.

Another complication is that obal doesn't use this script anyway and has duplicated the logic.